epoxy-processor/src/main/java/com/airbnb/epoxy/ModelViewProcessor.kt

```diff
@@ -354,7 +408,13 @@ internal class ModelViewProcessor(
     }
 
     private fun validateResetElement(resetMethod: Element): Boolean =
-            validateExecutableElement(resetMethod, OnViewRecycled::class.java, 0)
+        validateExecutableElement(resetMethod, OnViewRecycled::class.java, 0)
+
+    private fun validateVisibilityStateChangedElement(visibilityMethod: Element): Boolean =
+        validateExecutableElement(visibilityMethod, OnVisibilityStateChanged::class.java, 1)
+
+    private fun validateVisibilityChangedElement(visibilityMethod: Element): Boolean =
+        validateExecutableElement(visibilityMethod, OnVisibilityChanged::class.java, 4)
```

Comment:
`validateExecutableElement` was previously only used for parameters that could be anything, so only the count mattered, for this it is important that we give a clear warning if the types don't match the expected. Can you change `validateExecutableElement` to take an optional list of parameter types, and emit an error if the types don't match?